protocol/operator-fee-scalar-fix.md

```diff
@@ -0,0 +1,95 @@
+
+- [Operator Fee Scalar Fix](#operator-fee-scalar-fix)
+      - [Metadata](#metadata)
+- [Summary](#summary)
+- [Problem Statement + Context](#problem-statement--context)
+- [Proposed Solution](#proposed-solution)
+  - [Alternatives considered](#alternatives-considered)
+- [Risks & Uncertainties](#risks--uncertainties)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Operator Fee Scalar Fix
+
+#### Metadata
+
+Authors: @yuwen01.
+Created: July 16, 2025.
+
+# Summary
+
+This is the current operator fee formula.
+
+```
+operatorFee = operatorFeeConstant + operatorFeeScalar * gasUsed / 1e6
+```
+
+The `1e-6` multiplier on the `operatorFeeScalar` is not useful for operators as it results in a fee that is too low to be useful. This design doc proposes the following change to the operator fee formula.
+
+```
+operatorFee = operatorFeeConstant + operatorFeeScalar * gasUsed * 100
+```
+
+<!-- Most (if not all) documents should have a summary.
+While the length will likely be proportional to the length of the full document,
+the summary should be as succinct as possible. -->
+
+# Problem Statement + Context
+
+<!-- Describe the specific problem that the document is seeking to address as well
+as information needed to understand the problem and design space.
+If more information is needed on the costs of the problem,
+this is a good place to that information. -->
+
+This is the current operator fee formula.
+
+```
+operatorFee = operatorFeeConstant + operatorFeeScalar * gasUsed / 1e6
+```
+
+Where `operatorFeeScalar` is a `u32`, and `operatorFeeConstant` is a `u64`. However, for a 2.5 million gas transaction and a `operatorFeeScalar` of `u32::MAX_VALUE`, the operator fee will be `u32::MAX_VALUE * 2.5e6 / 1e6 ~= 10 gwei`. With an ETH price of 2.5k, this is 0.0000027 USD -- a negligible fee that is not useful for operators.
+
+# Proposed Solution
+
+<!-- A high level overview of the proposed solution.
+When there are multiple alternatives there should be an explanation
+of why one solution was picked over other solutions.
+As a rule of thumb, including code snippets (except for defining an external API)
+is likely too low level. -->
+
+We propose the following change to the operator fee formula.
+
+```
+operatorFee = operatorFeeConstant + operatorFeeScalar * gasUsed * 100
+```
+
+Where `operatorFeeScalar` and `operatorFeeConstant` remain `u32` and `u64` respectively.
+
+Examine the following to get a sense of the range of fees this formula supports. For the dollar cost column, we assume an ETH price of 2.5k.
+
+| Gas Used | `operatorFeeScalar` | `operatorFeeScalar * gasUsed * 100` | Dollar Cost |
+|----------|-------------------|----------------------------------|-------------|
+| 30,000,000 | 1 | 3 gwei | $0.0000075 |
+| 30,000,000 | 2^32 | 1.29e10 gwei | $32,250.00 |
+| 1,000,000 | 1 | 0.1 gwei | $0.00000025 |
+| 1,000,000 | 2^32 | 4.29e8 gwei | $1,072.50 |
```

Comment:
I think this table supports that `100` is a good factor, it seems to cover all imaginable ranges.

Reply:
It would also be interesting to add two more entries for a tx that only consumes, say, 21,000 gas.

Reply:
Let's add another row to this table with the minimum gas to get full confidence on this table.